kubeflow/trainer/rhai/utils.py

```
@@ -203,3 +260,64 @@ def apply_output_dir_uri_to_pod_overrides(
+    """
+    from kubernetes.client.rest import ApiException
+
+    try:
+        core_api.read_namespaced_secret(name=secret_name, namespace=namespace)
+    except ApiException as e:
+        if e.status == 404:
+            raise ValueError(
+                f"Secret '{secret_name}' not found in namespace '{namespace}'. "
+                "Please create the Data Connection secret or verify the "
+                "data_connection_name is correct."
+            ) from e
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)